packages/node-resolve/src/package/resolvePackageImportsExports.ts

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable no-await-in-loop */
+import resolvePackageTarget from './resolvePackageTarget';
+
+import { InvalidModuleSpecifierError } from './utils';
+
+/**
+ * Implementation of Node's `PATTERN_KEY_COMPARE` function
+ */
+function nodePatternKeyCompare(keyA: string, keyB: string) {
+  // Let baseLengthA be the index of "*" in keyA plus one, if keyA contains "*", or the length of keyA otherwise.
+  const baseLengthA = keyA.includes('*') ? keyA.indexOf('*') + 1 : keyA.length;
+  // Let baseLengthB be the index of "*" in keyB plus one, if keyB contains "*", or the length of keyB otherwise.
+  const baseLengthB = keyB.includes('*') ? keyB.indexOf('*') + 1 : keyB.length;
+
+  // if baseLengthA is greater, return -1, if lower 1
+  const rval = baseLengthB - baseLengthA;
+  if (rval !== 0) return rval;
+
+  // If keyA does not contain "*", return 1.
+  if (!keyA.includes('*')) return 1;
+  // If keyB does not contain "*", return -1.
+  if (!keyB.includes('*')) return -1;
+
+  // If the length of keyA is greater than the length of keyB, return -1.
+  // If the length of keyB is greater than the length of keyA, return 1.
+  // --> both keys do not contain * so baseLength of both is the key length, so this is already handled by `rval` above
```

Comment:
True, but if order is important here (I did not check the original algorithm), then length is now checked before the presence of `*` instead of after. Is this fine?

Reply:
Good catch @lukastaegert - both keys contain `*` here so I think we do need to still do the comparison as per the original algorithm, since the greater specificity when the bases are equal is taken to be the one with the more specific trailer.

Reply:
Good catch, asked myself why the spec has this useless condition, but I simply missed the not.

At this position both keys contain a `*` so we really need to use the full key length of both.
```suggestion
  // If the length of keyA is greater than the length of keyB, return -1.
  // If the length of keyB is greater than the length of keyA, return 1.
  return keyB.length - keyA.length
```